example-app/src/main/java/com/adyen/checkout/example/ui/card/compose/SessionsCardScreen.kt

```diff
@@ -0,0 +1,139 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.adyen.checkout.card.CardComponent
+import com.adyen.checkout.card.CardConfiguration
+import com.adyen.checkout.components.compose.AdyenComponent
+import com.adyen.checkout.components.compose.get
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.example.ui.card.SessionsCardComponentData
+import com.adyen.checkout.example.ui.card.SessionsCardViewModel
+import com.adyen.checkout.example.ui.compose.ResultContent
+
+@Composable
+internal fun SessionsCardScreen(
+    onBackPressed: () -> Unit,
+) {
+    Scaffold(
+        modifier = Modifier.windowInsetsPadding(WindowInsets.ime),
+        topBar = {
+            TopAppBar(
+                title = { Text(text = "Card component with sessions") },
+                navigationIcon = {
+                    IconButton(onClick = { onBackPressed() }) {
+                        Icon(Icons.Filled.ArrowBack, contentDescription = "Back")
+                    }
+                },
+            )
+        },
+    ) { innerPadding ->
+        MainContent(Modifier.padding(innerPadding))
+    }
+}
+
+@Suppress("DestructuringDeclarationWithTooManyEntries")
+@Composable
+private fun MainContent(
+    modifier: Modifier = Modifier,
+    viewModel: SessionsCardViewModel = hiltViewModel(),
+) {
+    Box(
+        modifier = modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        val uiState by viewModel.uiState.collectAsState()
+
+        val (cardConfiguration, isLoading, toastMessage, componentData, action, finalResult) = uiState
+
+        if (isLoading) {
+            CircularProgressIndicator()
+        }
+
+        if (toastMessage != null) {
+            Toast.makeText(LocalContext.current, toastMessage, Toast.LENGTH_SHORT).show()
+            viewModel.toastMessageConsumed()
+        }
+
+        if (finalResult != null) {
+            ResultContent(finalResult)
+        } else if (componentData != null) {
+            CardComponent(
+                configuration = cardConfiguration,
+                componentData = componentData,
+                action = action,
+                onActionConsumed = viewModel::actionConsumed,
+                modifier = Modifier.fillMaxSize(),
+            )
+        }
+    }
+}
+
+@Composable
+private fun CardComponent(
+    configuration: CardConfiguration,
+    componentData: SessionsCardComponentData,
+    action: Action?,
+    onActionConsumed: () -> Unit,
+    modifier: Modifier = Modifier,
```

Comment:
Nits: Usually modifier is the first argument in Composable functions. I think it would be good to keep it consistent.